src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1727,12 +1791,444 @@ def get_no_wait(self, **kwargs) -> bool:
+        if outbound_type != CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
+            dynamic_completion = True
+        # disable dynamic completion if the value is read from `mc`
+        dynamic_completion = dynamic_completion and not read_from_mc
+        if dynamic_completion:
+            outbound_type = CONST_OUTBOUND_TYPE_LOAD_BALANCER
+
+        # validation
+        # Note: The parameters involved in the validation are not verified in their own getters.
+        if enable_validation:
+            if outbound_type == CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
+                # Should not enable read_only for get_load_balancer_sku, since its default value is None, and it has
+                # not been decorated into the mc object at this time, only the value after dynamic completion is
+                # meaningful here.
+                if self.get_load_balancer_sku() == "basic":
+                    raise InvalidArgumentValueError(
+                        "userDefinedRouting doesn't support basic load balancer sku"
+                    )
+
+                if self.get_vnet_subnet_id() in ["", None]:
+                    raise RequiredArgumentMissingError(
+                        "--vnet-subnet-id must be specified for userDefinedRouting and it must "
+                        "be pre-configured with a route table with egress rules"
+                    )
+
+                load_balancer_profile = kwargs.get("load_balancer_profile")
+                if load_balancer_profile:
+                    if (
+                        load_balancer_profile.managed_outbound_i_ps or
+                        load_balancer_profile.outbound_i_ps or
+                        load_balancer_profile.outbound_ip_prefixes
+                    ):
+                        raise MutuallyExclusiveArgumentError(
+                            "userDefinedRouting doesn't support customizing a standard load balancer with IP addresses"
+                        )
+                else:
+                    if (
+                        self.get_load_balancer_managed_outbound_ip_count() or
+                        self.get_load_balancer_outbound_ips() or
+                        self.get_load_balancer_outbound_ip_prefixes()
+                    ):
+                        raise MutuallyExclusiveArgumentError(
+                            "userDefinedRouting doesn't support customizing a standard load balancer with IP addresses"
+                        )
+
+        return outbound_type
+
+    # pylint: disable=unused-argument
+    def get_network_plugin(self, enable_validation: bool = False, **kwargs) -> Union[str, None]:
+        # read the original value passed by the command
+        raw_value = self.raw_param.get("network_plugin")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        value_obtained_from_mc = None
+        if self.mc and self.mc.network_profile:
```

Comment:
add one more and ,and you will save L2054 to L2055.

Reply:
Changed in commit [524654c](https://github.com/Azure/azure-cli/pull/19458/commits/524654cc0c83db5b206ea2a30ec321e6632c8980).